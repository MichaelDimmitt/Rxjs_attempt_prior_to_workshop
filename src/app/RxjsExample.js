import React, { Component } from 'react';
import rxpng from '../assets/rxjs.png';
import '../assets/css/animation.css';
import '../assets/css/additionalStyling.css';
import { LandingPage } from './landingPage/LandingPage.js';
import { TestSuiteIntroduction } from './testSuiteIntroduction/TestSuiteIntroduction.js'
import { RequestManipulationRxjs } from './brokenPages/RequestManipulationRxjs.js'

// import {TestSuiteIntroduction} from './testSuiteIntroduction/OldTestSuiteIntroduction.js'

function determineComponents(currRoute) {
  {
    console.log(currRoute)
    switch (currRoute) {
      case '/': return (<LandingPage rxpng={rxpng}/>); break;
      case '/main_landing_page.html': return (<TestSuiteIntroduction rxpng={rxpng}/>); break;
      case '/brokenPages/requests.html': return (<RequestManipulationRxjs rxpng={rxpng}/>); break;
      // case '/main_rxjs_page.html': return (<RxjsRunner/>); break;
      default: return (<LandingPage rxpng={rxpng}/>);
    }
  }
}

class App extends Component {
  render() {
  let currRoute = window.location.href.substring(21)
    return (
      <div>
        {determineComponents(currRoute)}
      </div>
    );
  }
}

export default App;
