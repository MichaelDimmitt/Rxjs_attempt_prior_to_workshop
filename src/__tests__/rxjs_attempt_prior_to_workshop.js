// Copyright 2004-present Facebook. All Rights Reserved.

/* eslint-disable no-unused-vars */

import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import * as TestUtils from 'react-dom/test-utils';
import RxjsExample from '../app/RxjsExample.js';
import context from 'jest-plugin-context';

it('should render correctly', () => {
    const tree = renderer
      .create(<RxjsExample/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
});

describe('LandingPage section', () => {
  it('should dangerously be importing Rxjs and able to handle an rxjs block', () => {

    //    let foo = function () { Rx.Observable.of(`Hello World`)
    //      .subscribe(result => console.log(result)); };
    //
    //    const rxjsExample = TestUtils.renderIntoDocument(
    //      <RxjsExample check={foo} />
    //    const rxjsExampleNode = ReactDOM.findDOMNode(rxjsExample);
    //  })
    //  context('Rxjs Subscribing to images', () => {
    //    it('should render the image after Rxjs subscription.', () => {
    //      // Render the landing page. With the image subscribed.
    //      const checkboxNode = ReactDOM.findDOMNode(checkbox);
    //
    //      // Verify that it's Off by default
    //      expect(checkboxNode.textContent).toEqual('Off');
    //
    //      // Simulate a click and verify that it is now On
    //      TestUtils.Simulate.change(
    //        TestUtils.findRenderedDOMComponentWithTag(checkbox, 'input')
    //      );
    //      expect(checkboxNode.textContent).toEqual('On');
    //    });
  });
})

 describe('brokenPages section', () => {
   // context('Request', () => {
   // });
   // context('NodeCalculationRxjs', () => {
   // });
   // context('GameRxjs', () => {
   // });
 })
