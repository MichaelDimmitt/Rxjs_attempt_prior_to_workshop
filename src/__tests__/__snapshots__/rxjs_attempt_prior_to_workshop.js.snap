// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div>
    <img
      alt=""
      id="loading"
      src="rxjs.png"
    />
    <div
      className="reposition"
    >
      <h1>
        RxJS the way of the future.
      </h1>
      <p>
        examples in slim supply.
      </p>
      <a
        href="main_landing_page.html"
      >
        click here
      </a>
    </div>
  </div>
</div>
`;

exports[`should render correctly 1`] = `
<div>
  <div>
    <img
      alt=""
      id="loading"
      src="rxjs.png"
    />
    <div
      className="reposition"
    >
      <h1>
        RxJS the way of the future.
      </h1>
      <p>
        examples in slim supply.
      </p>
      <a
        href="main_landing_page.html"
      >
        click here
      </a>
    </div>
  </div>
</div>
`;
